### PR TITLE
[Review] Request from 'aduffeck' @ 'SUSE/machinery/review_150625_improve_compare_packages_output'

### PR DESCRIPTION
**Pull request by aduffeck**
Please review the following changes:
  * 5937d08 Refactor Renderer API
  * f0eb2b0 Change PackagesScope#compare_with to consider changed packages
  * c468536 Add an (optional) field for changed elements to #compare_with
  * 9b917ab Fix typo
  * 5b0a26d Do not consider release attribute when the version has changed
  * 88397b9 Fix test, it was not testing what it said it tests
  * c92fdd9 Fix hound issues
  * a93f99d Expect if clauses spanning multiple lines to be indented, not aligned
  * e08356d Handle corner cases when comparing packages scopes
  * 7246e87 Add changelog entry for improved package comparison output
  * f8ca83b Add documentation for the comparison rendering process
  * 2e9b232 Add section to packages comparisons for same-but-different packages
  * 609d581 Extract high-level comparison structure into #do_render_comparison
  * f52f4fb Allow for customizing the rendering of comparison results

### CODE REVIEW

File: lib/compare_task.rb [hunk 1]
```diff
@@ -74,17 +74,18 @@ def render_comparison(description1, description2, scopes, options = {})
...
           "common",
           store,
-          scope => comparison[2]
+          scope => comparison[3]
         )
 
         output += Renderer.for(scope).render_comparison(
           partial_description1,
           partial_description2,
+          comparison[2],
```
Comment (cornelius) [inline lib/compare_task.rb]: This is inconsistent in terms of the API. It would be more logical to also wrap the changed elements in a description, so that the call would use the arguments `partial_description1, partial_description2, partial_description_changed, partial_description_common`.

File: lib/renderer.rb [hunk 2]
```diff
@@ -94,54 +117,85 @@ def render(system_description, options = {})
...
-  end
-
-  def render_comparison(description1, description2, description_common, options = {})
+  # Renders the result of a comparison of two system descriptions. The arguments to that method
+  # are three SystemDescriptions:
+  #
+  # [description1] contains all elements that are only present in description1
+  # [description2] contains all elements that are only present in description2
+  # [description_common] contains all elements that are present in both descriptions
+  #
```
Comment (cornelius) [inline lib/renderer.rb]: The API documentation also needs to document the argument with the changed elements.

**Comment by cornelius** [comment]
Maybe it would be good to change the return value of the `compare_with` method to be a dedicated object with named accessors to the different parts of the comparison result. Just having a plain array makes the calling code not very easy to read as you need to know on which position which part is.

**Comment by cornelius** [comment]
I think introducing the new naming convention for the methods to be overridden by the specific renderers makes sense. There still might be ways how to improve that, but at least the basic structure is clear now.
